WordPress/gutenberg
# Close overlay menu when clicking an anchor link

## What?

This PR adds an event Listener to toggle the navigation block overlay menu on clicking links inside the overlay.

Fixes #39585

## Why?

Navigating to anchor links in the navigation block in an overlay menu does not automatically close the overlay. This requires manually closing the modal to view the link's content.

## How?

Since we are using [Micromodal](https://github.com/ghosh/Micromodal) to implement the overlay menu, am using the `MicroModal.close()` method to toggle the modal on clicking the navigation link. My initial idea was to use the `data-micromodal-close` custom attribute which Micromodal uses as a trigger to close the modal but I couldn't get it to work reliably. The overlay would close but sometimes the browser would not navigate to the correct anchor.

## Testing Instructions





1. Open a post or page.
2. Create a navigation menu and add an anchor link.
3. View the post/page on the frontend.
4. Make your browser screen smaller, below 600px.
5. Open the menu and click the anchor link.
7.  Observe that the overlay menu modal closes and page navigates to the anchor link's content. Try also testing by changing the navigation block overlay menu display  setting to `off` , `always` or `mobile`.


## Screenshots or screencast 

#### Before
https://user-images.githubusercontent.com/276316/159422355-a6ad6d65-e866-4874-94fa-9fa6f4beca87.mov 

### After
https://user-images.githubusercontent.com/276316/159422443-e9817778-5499-4590-bd9d-f39abf81040c.mov


Fixes #39585
